zokrates-worker/src/index.mjs

```diff
@@ -8,10 +10,80 @@ import logger from './utils/logger.mjs';
 
 const {
   MPC: { MPC_PARAMS_URL },
+  DEPLOYMENT_FILES_URL: { DEFAULT_CIRCUIT_FILES_URL },
 } = config;
 
+const { ETH_NETWORK, CIRCUIT_FILES_URL } = process.env;
+
+const finished = promisify(stream.finished);
+
+const downloadFile = async (fileUrl, outputLocationPath) => {
+  const writer = fs.createWriteStream(outputLocationPath);
+  return axios({
+    method: 'get',
+    url: fileUrl,
+    responseType: 'stream',
+  }).then(response => {
+    response.data.pipe(writer);
+    return finished(writer); // this is a Promise
+  });
+};
+
+const checkCircuitsOutput = async () => {
+  let env = '';
+  switch (ETH_NETWORK) {
+    case 'goerli':
+      env = 'testnet';
+      break;
+    case 'mainnet':
+      env = 'production';
+      break;
+    default:
+      env = '';
+  }
+
+  if (env) {
+    const baseUrl = CIRCUIT_FILES_URL
+      ? `${CIRCUIT_FILES_URL}`
+      : `${DEFAULT_CIRCUIT_FILES_URL}/${env}`;
+    const url = `${baseUrl}/proving_files/hash.txt`;
+    const outputPath = `./output`;
+    const circuits = ['deposit', 'double_transfer', 'single_transfer', 'withdraw'];
+
+    const res = await axios.get(url); // get all circuit files
+    const files = res.data.split('\n');
+
+    logger.info(`Downloading output files from ${url}...`);
+
+    await Promise.all(
+      files.map(async f => {
+        if (f) {
+          const filename = f.split('  ')[1];
+          const circuit = circuits.find(c => filename.includes(c));
+
+          if (!fs.existsSync(`${outputPath}/${circuit}`)) {
+            fs.mkdirSync(`${outputPath}/${circuit}`);
+          }
+
+          try {
+            await downloadFile(
+              `${baseUrl}/proving_files/${circuit}/${f.split('  ')[1]}`,
+              `${outputPath}/${circuit}/${f.split('  ')[1]}`,
+            );
+          } catch (e) {
+            console.error(`ERROR downloading ${f.split('  ')[1]}`);
```

Comment:
The same as the comment I added before, if the contract strictly required for the app to run well, we should let the error go up.